### PR TITLE
[DefaultType] fix rigidcoord compilation

### PR DESCRIPTION
**Pull request by alxbilger**
Replaces #3360 as the pull request is blocked.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by damienmarchal** [comment]
Ok to me.

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h [hunk 1]
```diff
@@ -267,15 +267,17 @@ class RigidCoord<3,real>
...
         orientation.toMatrix(m);
-        m[0][3] = (typename Mat::Real)center[0];
-        m[1][3] = (typename Mat::Real)center[1];
-        m[2][3] = (typename Mat::Real)center[2];
+    }
+
+    /// Write to the given 4x4 matrix
+    void toMatrix( sofa::type::Mat<4,4,real>& m) const
+    {
+        toHomogeneousMatrix(m);
```
Comment (fredroy) [inline Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h:280]: It seems that the translation part has been forgotten no ?

**Review by fredroy — CHANGES_REQUESTED** [review]

File: Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h (hunk 1 above)
Comment (alxbilger) [reply Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h:280]: The `toHomogeneousMatrix` function is just bellow. Do you see something wrong?

Comment (fredroy) [reply Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h:280]: In the two versions, the translation part is not set

**Comment by damienmarchal** [comment]
@fredroy  can you validate that the requested change has been done ?

**Comment by fredroy** [comment]
> @fredroy can you validate that the requested change has been done ?

More or less, as I did not see that `toHomogeneousMatrix` was considering translation, and the Mat3x3 cannot store the translation. Anyway, the mat3x3 version should be renamed toRotationMatrix as the two versions do something a bit different.
But that is not the point of the PR itself (fix compilation)

**Review by fredroy — APPROVED** [review]

**Pull request merged**